migrations/20230726151100_add_pizza_service_tables.sql

```diff
@@ -0,0 +1,54 @@
+------------------------------------
+-- Pizza oven "baked" repos table --
+-------------------------------------
+create table if not exists public.baked_repos
+(
+  id bigint not null generated by default as identity ( increment 1 start 1 minvalue 1 maxvalue 9223372036854775807 cache 1 ),
+  clone_url character varying(255) collate pg_catalog."default" not null,
+
+  -- dynamic columns
+  constraint baked_repos_pkey primary key (id)
+)
+
+tablespace pg_default;
+
+-- indexes for baked repos
+create index if not exists baked_repos_idx_clone_url on baked_repos (clone_url);
+
+
+-------------------------------------
+-- Pizza oven commit authors table --
+-------------------------------------
+create table if not exists public.commit_authors
+(
+  id bigint not null generated by default as identity ( increment 1 start 1 minvalue 1 maxvalue 9223372036854775807 cache 1 ),
+  commit_author_email character varying(255) collate pg_catalog."default" not null,
+
+  -- dynamic columns
+  constraint commit_authors_pkey primary key (id)
+)
+
+tablespace pg_default;
+
+-- indexes for baked repos
+create index if not exists commit_authors_idx_commit_author_email on commit_authors (commit_author_email);
+
+--------------------------------
+-- Pizza oven indexed commits --
+--------------------------------
+
+create table if not exists public.commits (
+  id bigint not null generated by default as identity ( increment 1 start 1 minvalue 1 maxvalue 9223372036854775807 cache 1 ),
+  baked_repo_id bigint not null references public.baked_repos (id) on delete cascade on update cascade,
+  commit_author_id bigint not null references public.commit_authors(id) on delete cascade on update cascade,
+  commit_hash character varying(255) collate pg_catalog."default" not null,
+  commit_date timestamp with time zone not null default now(),
```

Comment:
We usually store timestamp without the timezone, so when its displayed to the user, there local timezone is used.

```suggestion
  commit_date timestamp without time zone not null default now(),
```

Reply:
I had originally considered using the date/time directly from the commits without the timestamp in the database, but that created some really weird behavior in the pizza service when attempting to get the "latest" commits by date. The wrong commit would be returned since commits are timestamped by the individual commit author (which may have originated from within any timezone),

Attempting to drop them into the database _without_ a timestamp results in the unexpected ordering by date. So, my solution in the pizza service was the convert them during indexing to UTC:

https://github.com/open-sauced/pizza/blob/e99d988bbc30d3fc9fa4e4a6cbb0d789a5298d76/pkg/server/server.go#L169

and enter them in the database with the UTC timestamp explicitly.